pulp_container/app/viewsets.py

```diff
@@ -113,8 +113,51 @@ class Meta:
             "name": NAME_FILTER_OPTIONS,
         }
 
-
-class TagViewSet(ReadOnlyContentViewSet):
+from collections import namedtuple
+
+Repo = namedtuple("Object" , "type perm")
+mirror_repo = Repo("container.container", "container.view_containerrepository")
+push_repo = Repo("container.container-push", "container.view_containerpushrepository")
+
+class ContainerQuerySetViewSet(viewsets.GenericViewSet):
+    def get_queryset(self):
+        qs = super().get_queryset()
+        has_model_push_repo = self.request.user.has_perm(push_repo.perm)
+        has_model_repo = self.request.user.has_perm(mirror_repo.perm)
+        if has_model_push_repo and has_model_repo:
+            return qs
+        #if has_model_repo:
```

Comment:
I commented this out because it might get false-positive results.
If a user has model level perms for push-repo but only some object-level perms on mirror-repo, we won't show any mirror-repo results.

Reply:
this was an attempt to not go trough lines 140+